tests/unit_tests/service/test_interface.py

```diff
@@ -436,6 +440,32 @@ def test_setup(mock_stomp: MagicMock):
     assert interface.worker()._ctx is not None
     assert interface.context().run_engine.scan_id_source == interface._update_scan_num
 
+    clear_path_provider()
+    interface.teardown()
+
+
+@patch("blueapi.service.interface.StompClient")
+def test_setup_without_path_provider_with_numtracker_makes_start_document_provider(
+    mock_stomp: MagicMock,
+):
+    interface.teardown()
+
+    conf = ApplicationConfig(
+        env=EnvironmentConfig(
+            metadata=MetadataConfig(instrument="p46", instrument_session="ab123")
+        ),
+        numtracker=NumtrackerConfig(),
+    )
+    interface.set_config(conf)
+    interface.setup(conf)
+
+    path_provider = get_path_provider()
+
+    assert isinstance(path_provider, StartDocumentPathProvider)
+    assert interface.worker()._ctx is not None
```

Comment:
Could: I would prefer not to have private member access unless you think it adds a lot of value to the test